src/inference/src/dev/core_impl.cpp

```diff
@@ -592,6 +592,15 @@ ov::Plugin ov::CoreImpl::get_plugin(const std::string& pluginName) const {
             so = ov::util::load_shared_object(desc.libraryLocation.c_str());
             std::shared_ptr<ov::IPlugin> plugin_impl;
             reinterpret_cast<ov::CreatePluginFunc*>(ov::util::get_symbol(so, ov::create_plugin_function))(plugin_impl);
+            const auto& plugin_name = plugin_impl->get_device_name();
+
+            // Check that device plugin name is the same as requested for HW plugins
+            if (!plugin_name.empty() && !is_virtual_device(plugin_name)) {
+                OPENVINO_ASSERT(deviceName.find(plugin_name) != std::string::npos,
+                                deviceName,
+                                " device loaded with incorrect name: ",
+                                plugin_name);
```

Comment:
maybe we need to write a message that file "libraryLocation" is used for device "deviceName", while it contains implementation for "plugin_name" ?